### ISSUE TITLE
imperial units silently removed from ele tag

### ISSUE THREAD

**Issue by SK53** [openstreetmap/iD#5658]
I was adding [a peak](https://www.openstreetmap.org/node/6166375481#map=19/55.23718/-6.93624) from old OOC maps which have all elevation data in feet. I therefore entered 561 ft in the elevation field, so was somewhat surprised to find it rendered as 561 m. Testing shows that iD silenty strips any non-numeric text (including ",") from what is entered. Although ele:imperial is probably the most widely used tag for non-metric units, the ele tag with an indication of units (e.g., ft, feet, or "'") has always been in use. I would expect at least a warning, because there is a big difference between 561 feet & metres.

**Comment by bhousel** [comment]
Oh yeah, good catch @SK53 
This is like #5438 / #5570 

iD tries too hard to parse the numeric fields as numbers.

I think we probably need to take a first pass with a regex to decide whether the field is strictly numeric before trying to parse it.

**Comment by boothym** [comment]
Entering "561 ft" in the ele field is a tagging mistake. ele:imperial is used only once on the map - ele:ft is the more widely used tag (2k+).

It would be good if that could be picked up by iD - highlighting the field with a red border or similar if it's not strictly numeric. Basically input type="number" but without the browser's up/down controls, like the website field with type="url":
![osm](https://user-images.githubusercontent.com/16965772/50495140-29057c00-0a1f-11e9-813b-bd0aff4ca126.JPG)

**Comment by systemed** [comment]
> Entering "561 ft" in the ele field is a tagging mistake

Nope. Space-separated units have been used in OSM since time immemorial and most major clients parse them, such as [osm2pgsql](https://github.com/openstreetmap/osm2pgsql/blob/932ded7f191bad5994b67ff60038ffe211d00efe/table.cpp#L526), mkgmap, [OSRM](https://github.com/Project-OSRM/osrm-backend/blob/714719c3776c5985a8881cba32f48f8a4de5a70a/profiles/lib/measure.lua#L12), [Graphhopper](https://github.com/graphhopper/graphhopper/blob/5c980b4627578626268873f9424de35188187441/core/src/main/java/com/graphhopper/routing/util/DataFlagEncoder.java#L485) etc.

**Comment by bhousel** [comment]
Right, regardless of whether it happens to be a "tagging mistake" in the "ele" field - this is something iD needs to support in all the numeric fields.  

I've actually been thinking of adding a "default units" property to each field that can have such a thing.  There are a lot of them.  But right now we include the default unit in the field label _sometimes_, and in the placeholder _sometimes_, but none of this is done consistently, and I'm sure it makes a mess for translators.

https://github.com/openstreetmap/iD/blob/master/data/presets/fields/fire_hydrant/pressure.json
https://github.com/openstreetmap/iD/blob/master/data/presets/fields/generator/output/electricity.json

**Comment by 1ec5** [comment]
The `ele` key dates back to a simpler time when a lot of people thought the `123 unit` format would be too unstructured for data consumers to be willing to parse. But that was before a whole host of keys started expecting that format, not to mention `ref` on ways and mini languages like `opening_hours` and `turn:lanes`. As things stand, `ele` [stands out](https://wiki.openstreetmap.org/wiki/Talk:Key:ele#Feet_and_inches) as a rare key that doesn’t accept units. I think that should change, but there should be some wider discussion so that data consumers aren’t caught unawares. It’s entirely possible that some renderers have implemented the same number parsing as iD and would misleadingly interpret `561 ft` as 561&nbsp;meters.

ideditor/schema-builder#15 requests a new field type for measured values, so that the Elevation field could display a dropdown menu for selecting meters or feet-and-inches.

**Comment by 1ec5** [comment]
> As things stand, `ele` [stands out](https://wiki.openstreetmap.org/wiki/Talk:Key:ele#Feet_and_inches) as a rare key that doesn’t accept units. I think that should change, but there should be some wider discussion so that data consumers aren’t caught unawares.

https://wiki.openstreetmap.org/wiki/Proposed_features/ele_with_units

**Comment by 1ec5** [comment]
“(Meters)” was appended to the field name in openstreetmap/id-tagging-schema#95, which may decrease the likelihood of a user accidentally running into this trap, but regardless the field remains somewhat unintuitive in the United States.
